arborist/auth.go

```diff
@@ -115,7 +116,7 @@ func authorize(request *AuthRequest) (*AuthResponse, error) {
 	}
 
 	// run authorization query
-	rows, err := request.stmts.Query(
+	rv, err := _authorize(request.stmts, exp, args,
```

Comment:
I think I would be in favor of breaking this function down into a separate `authorizeUser` and `authorizeClient` functions (which both take an `*AuthRequest`)

Reply:
Fixed in a way ;)